sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -0,0 +1,730 @@
+ * @pre \p client must not be `NULL`.
+ * @pre \p request_id must be a valid, non-empty #az_span.
+ * @pre \p mqtt_topic must not be `NULL`.
+ * @pre \p mqtt_topic_size must be greater than 0.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic was retrieved successfully.
+ */
+AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_twin_document_get_publish_topic(
+    az_iot_pnp_client const* client,
+    az_span request_id,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
+{
+  return az_iot_hub_client_twin_document_get_publish_topic(
+      &(client->_internal.iot_hub_client),
+      request_id,
+      mqtt_topic,
+      mqtt_topic_size,
+      out_mqtt_topic_length);
+}
+
+/**
+ * @brief Gets the MQTT topic that is used to submit a Plug and Play Property PATCH request.
+ * @note The payload of the MQTT publish message should contain a JSON document
+ *         formatted according to the DTDL specification.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] request_id The request id.
+ * @param[out] mqtt_topic A buffer with sufficient capacity to hold the MQTT topic. If
+ *                        successful, contains a null-terminated string with the topic that
+ *                        needs to be passed to the MQTT client.
+ * @param[in] mqtt_topic_size The size, in bytes of \p mqtt_topic.
+ * @param[out] out_mqtt_topic_length __[nullable]__ Contains the string length, in bytes, of
+ *                                                  \p mqtt_topic. Can be `NULL`.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p request_id must be a valid, non-empty #az_span.
+ * @pre \p mqtt_topic must not be `NULL`.
+ * @pre \p mqtt_topic_size must be greater than 0.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic was retrieved successfully.
+ */
+AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_twin_patch_get_publish_topic(
+    az_iot_pnp_client const* client,
+    az_span request_id,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
+{
+  return az_iot_hub_client_twin_patch_get_publish_topic(
+      &(client->_internal.iot_hub_client),
+      request_id,
+      mqtt_topic,
+      mqtt_topic_size,
+      out_mqtt_topic_length);
+}
+
+/**
+ * @brief Append the necessary characters to a reported property JSON payload belonging to a
+ * subcomponent.
+ *
+ * The payload will be of the form:
+ *
+ * @code
+ * "reported": {
+ *     "<component_name>": {
+ *         "__t": "c",
+ *         "temperature": 23
+ *     }
+ * }
+ * @endcode
+ *
+ * @note This API should be used in conjunction with
+ * az_iot_pnp_client_twin_property_end_component().
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in,out] json_writer The #az_json_writer to append the necessary characters for an IoT Plug
```

Comment:
Anywhere we use [in,out] and json_reader/json_writer parameters, change them to `ref_json_reader`/`ref_json_writer` (and vice versa), both in source and header files.